src/routes/web.php

```diff
@@ -58,3 +61,17 @@
     SigninController::class,
     'login'
 ]);
+
+Route::get('/create-post', function() {
+    $userPosts = UserPost::all();
+    $postMedia = PostMedia::all();
+    return view('components.create-post', [
+        'userPosts' => $userPosts, 
+        'postMedia' => $postMedia
+    ]);
+});
```

Comment:
Im not sure if this is correct but If you want to display posts, you need to create another route and handle that in the controller. 

This route is for "GET" method but your route is for creating post, creating posts is for POST method